RepositoryCompiler/CodeModel/CodeParsers/CSharp/CaDETClassMetricCalculator.cs

```diff
@@ -123,5 +124,43 @@ private int GetNumberOfMethodsDeclared(CaDETClass parsedClass)
         {
             return parsedClass.Members.Count(method => method.Type.Equals(CaDETMemberType.Method));
         }
+
+        /// <summary>
+        /// CC - Class cohesion
+        /// DOI: 10.14257/ijmue.2014.9.6.35
+        /// </summary>
+        private double? GetCCClassCohesion(CaDETClass parsedClass)
+        {
+            IEnumerable<CaDETMember> memberMethods = parsedClass.Members.Where(m => m.Type != CaDETMemberType.Property);
+            IEnumerable<CaDETMember> memberProperties = parsedClass.Members.Where(m => m.Type == CaDETMemberType.Property);
+            double fieldsCount = parsedClass.Fields.Count + memberProperties.Count();
+
+            if (fieldsCount == 0 || memberMethods.Count() == 0)
+            {
+                return 0;
+            }
+
+            double cv = CalculateCohesionForEveryVariable(parsedClass, memberMethods, memberProperties);
+
+            return Math.Round(cv / fieldsCount, 2);
+        }
+
+        private static double CalculateCohesionForEveryVariable(CaDETClass parsedClass, IEnumerable<CaDETMember> memberMethods, IEnumerable<CaDETMember> memberProperties)
```

Comment:
Variable is an odd name. Since we are looking for a common name for fields and properties, I googled this article https://stackoverflow.com/questions/26671085/is-there-a-common-word-in-use-for-fields-and-properties.

Since the term "Member" is used to encapsulate methods, fields, and properties, a comment suggested we call it "DataMember" and I like this.